src/main/java/org/usfirst/frc4048/common/swervev2/type/GenericSwerveModule.java

```diff
@@ -0,0 +1,65 @@
+package org.usfirst.frc4048.common.swervev2.type;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import org.usfirst.frc4048.common.swervev2.components.GenericEncodedSwerve;
+import org.usfirst.frc4048.common.util.Gain;
+import org.usfirst.frc4048.common.util.PID;
+
+public class GenericSwerveModule implements StatedSwerve, PidSwerve {
```

Comment:
Some javadoc on the Generic* classes would be nice.

Reply:
I am not sure what the convention is when writing Javadocs. What are some things I should include?